src/compiler/diagnosticMessages.json

```diff
@@ -7356,5 +7356,13 @@
     "A 'return' statement cannot be used inside a class static block.": {
         "category": "Error",
         "code": 18041
+    },
+    "An interface cannot extend a primitive type like '{0}'. An interface can only extend named object types.": {
+        "category": "Error",
+        "code": 18042
+    },
+    "A class cannot extend a primitive type like '{0}'. A class can only extend named object types.": {

```

Comment:
While you're adding class-specific errors, a separate error for when classes *implement* primitive types would be nice.